src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Helpers.cs

```diff
@@ -0,0 +1,145 @@
+            id: "SYSLIB1101",
+            title: new LocalizableResourceString(nameof(SR.PropertyNotSupportedTitle), SR.ResourceManager, typeof(FxResources.Microsoft.Extensions.Configuration.Binder.SourceGeneration.SR)),
+            messageFormat: new LocalizableResourceString(nameof(SR.PropertyNotSupportedMessageFormat), SR.ResourceManager, typeof(FxResources.Microsoft.Extensions.Configuration.Binder.SourceGeneration.SR)),
+            category: GeneratorProjectName,
+            defaultSeverity: DiagnosticSeverity.Info,
+            isEnabledByDefault: true);
+
+        // Unlike sourcegen warnings, exception messages should not be localized so we keep them in source.
+        private static class ExceptionMessages
+        {
+            public const string TypeNotSupported = "Unable to bind to type '{0}': '{1}'";
+        }
+
+        private static class Literal
+        {
+            public const string configuration = nameof(configuration);
+            public const string element = nameof(element);
+            public const string enumValue = nameof(enumValue);
+            public const string key = nameof(key);
+            public const string obj = nameof(obj);
+            public const string originalCount = nameof(originalCount);
+            public const string path = nameof(path);
+            public const string section = nameof(section);
+            public const string services = nameof(services);
+            public const string stringValue = nameof(stringValue);
+            public const string temp = nameof(temp);
+
+            public const string Add = nameof(Add);
+            public const string Any = nameof(Any);
+            public const string Bind = nameof(Bind);
+            public const string BindCore = nameof(BindCore);
+            public const string Configure = nameof(Configure);
+            public const string CopyTo = nameof(CopyTo);
+            public const string ContainsKey = nameof(ContainsKey);
+            public const string Count = nameof(Count);
+            public const string GeneratedConfigurationBinder = nameof(GeneratedConfigurationBinder);
+            public const string Get = nameof(Get);
+            public const string GetChildren = nameof(GetChildren);
+            public const string GetSection = nameof(GetSection);
+            public const string HasValue = nameof(HasValue);
+            public const string Length = nameof(Length);
+            public const string Parse = nameof(Parse);
+            public const string Resize = nameof(Resize);
+            public const string TryGetValue = nameof(TryGetValue);
+            public const string TryParse = nameof(TryParse);
+            public const string Value = nameof(Value);
+        }
+
+        private static class NotSupportedReason
+        {
+            public const string AbstractOrInterfaceNotSupported = "Abstract or interface types are not supported";
```

Comment:
should those be in the resource files?

Reply:
If these are meant to be exception messages in generated code, we generally don't resource these.

Reply:
It's hard to tell from the message, it looks like it might be produced error